src/Common/src/Interop/OSX/System.Security.Cryptography.Native.Apple/Interop.Ssl.cs

```diff
@@ -388,6 +390,12 @@ internal enum TlsCipherSuite
             int cbTargetName,
             out int osStatus);
 
+        [DllImport(Interop.Libraries.AppleCryptoNative, EntryPoint = "AppleCryptoNative_SslCtxSetAlpnProtos")]
+        internal static extern int SslCtxSetAlpnProtos(SafeSslHandle ctx, IntPtr protos, int len, out int osStatus);
```

Comment:
Instead of IntPtr and `fixed` this can be `ref byte` and you can call it with `ref buffer[0]` (provided buffer.Length isn't 0)

Reply:
This is what we do for OpenSSL. (src/Interop/Unix/System.Security.Cryptography.Native/Interop.SslCtx.cs) 
Should I change that as well? It seems reasonable to keep things consistent.
(This also applied to some comments bellow)

Reply:
If it was just changing the P/Invoke target then my normal tidiness heuristics would say that's a different cleanup change.

But since it looks like the ConvertAlpnProtocolListToByteArray method was a copy/paste, it should probably be factored out into shared source (vs copy/paste), and since you're now editing Interop.SslCtx.cs anyways you may as well change the P/Invoke :smile:.

Reply:
> this can be `ref byte`

We should be using `ref byte` for unsafe buffers. It is not capturing the intent. It is more appropriate to use use `byte*`.